### PR TITLE
Print `declare` and `export` keywords for nested namespace

### LINKED ISSUE #15207: `declare` keyword not formatted for a particular namespace name
**Prettier 3.0.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATOYA2BDATnAAimwFs4BnAB2zEO3JlwgAkAVAWQBkA6AKQGUAGgWABfEABoQESjACW0cslB4mAdwAKeBEpTZMa7AE8lUgEa4aAazgx+pOJzlQ4yAGb7ycc5bA271GDOAObIjACu3iBeJHJhuJFS5CGYcACK4RDw7p5RAFbkAB78KemZ2UgemF5SAI7lcBpMlLog9AC0LnDoqJIgjNhymCEAwhAkJNjIbZiYfclQwakAgjCMcmbh8BpwuE4uOdVRABYwJJgA6sdy8FQ0cPw6N3IAbjdG02DkpiAvkQCSUHQsH4YFwclkyyB-BgRlShxqIEoTC8F0slGmyIouxerikzi8uBgTWwwUmCKi1FwhOmsMoFDBEJgfWRzhgFzkqBgx2QAA4AAxSfD1OT4ElkqaVXJSGDYMwcrk8pAAJik4S8rDluiqiLgJDM3R6nGwi3CpLgADEILhJmsQtNsFsICBRKIgA)

```sh
# Options (if any):
```

**Input:**

```jsx
declare namespace astroHTML.JSX {}
```

**Output:**

```jsx
namespace astroHTML.JSX {}
```

**Expected behavior:**

```jsx
declare namespace astroHTML.JSX {}
```

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #15207

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,8 +38,8 @@
     "@iarna/toml": "2.2.5",
     "@prettier/is-es5-identifier-name": "0.2.0",
     "@prettier/parse-srcset": "3.1.0",
-    "@typescript-eslint/typescript-estree": "6.3.0",
-    "@typescript-eslint/visitor-keys": "6.3.0",
+    "@typescript-eslint/typescript-estree": "6.4.0",
+    "@typescript-eslint/visitor-keys": "6.4.0",
```
Comment (fisker) [inline package.json:42]: Can you upgrade `@typescript-eslint/eslint-plugin` too? So the dependencies doesn't get duplicated.

File: tests/format/typescript/module/namespace_nested.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace abc4.def {}
+
+declare namespace abc5.def.ghi {}
+
+export declare namespace abc2.def.ghi {}
+
+export namespace abc3.def.ghi {}
+
+namespace abc4.def.ghi {}
```
Comment (fisker) [inline tests/format/typescript/module/namespace_nested.ts:15]: Can you also replace `namespace` with `module` keyword, and test them?

Comment (sosukesuzuki) [reply tests/format/typescript/module/namespace_nested.ts:15]: done [4c60892](https://github.com/prettier/prettier/pull/15249/commits/4c6089266ee192cd8324180c7e72cc32ab75f3ca)

File: package.json (hunk 1 above)
Comment (sosukesuzuki) [reply package.json:42]: done [5cdb8a6](https://github.com/prettier/prettier/pull/15249/commits/5cdb8a6ae8eeffc5af6600ceb0d29f6e0bee1f74)

**Review by fisker — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
This is regression fix so I'll release patch after merging this.

**Pull request merged**